docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+
+```graphql
+query {
+  allPost {
+    nodes {
+      id
+      // highlight-start
+      author {
+        name
+      }
+      remoteImage {
+        id
+      }
+      // highlight-end
+    }
+  }
+}
+```
+
+### Using data from the source plugin in a site
+
+In the `example-site`, data is now available to be queried from pages.
+
+Add a file at `example-site/src/pages/index.js` and copy in the following code into it:
+
+```javascript:title=example-site/src/pages/index.js
+import React from "react"
+import { graphql } from "gatsby"
+import Img from "gatsby-image"
+
+export default ({ data }) => (
+  <div
+    style={{
+      padding: 32,
+    }}
+  >
+    <h1>Posts</h1>
+    <section
+      style={{
+        display: `grid`,
+        gridTemplateColumns: `repeat( auto-fit, minmax(250px, 1fr) )`,
+        gridGap: 16,
+        justifyContent: "space-between",
+      }}
+    >
+      {data.allPost.nodes.map(post => (
+        <div
+          style={{
+            display: `flex`,
+            flexDirection: `column`,
+            justifyContent: `space-between`,
+            padding: 16,
+            border: `1px solid #ccc`,
+            borderRadius: 8,
+          }}
+        >
+          <h2>{post.slug}</h2>
+          <span>By: {post.author.name}</span>
+          <p>{post.description}</p>
+          <Img
+            fluid={post.remoteImage.childImageSharp.fluid}
+            alt={post.imgAlt}
+            style={{
+              maxHeight: 300,
+            }}
+          />
+        </div>
+      ))}
+    </section>
+  </div>
+)
+
+export const query = graphql`
+  {
+    allPost {
+      nodes {
+        id
+        slug
+        description
+        imgAlt
+        author {
+          id
+          name
+        }
+        remoteImage {
+          id
+          childImageSharp {
+            id
+            fluid {
+              ...GatsbyImageSharpFluid
+            }
+          }
+        }
+      }
+    }
+  }
+`
+```
+
+This code uses a [page query](/docs/page-query/) to fetch all posts and provide them to the component in the `data` prop at build time. The posts are looped through and rendered to the DOM.
+
+### Using plugin options to customize plugin usage
+
+You can pass options into a plugin through the `gatsby-config.js`. Update the code where your plugin is installed in the `example-site`, changing it from a string, to an object with a `resolve` and `options` key.
+
+```javascript:title=example-site/gatsby-config.js
+module.exports = {
+  plugins: [
+    // highlight-start
+    {
+      resolve: require.resolve(`../source-plugin`),
+      options: {
+        previewMode: true,
+      },
+    },
+    // highlight-end
+    `gatsby-plugin-sharp`,
+    `gatsby-transformer-sharp`,
+  ],
+}
+```
+
+Now you can access the second argument on Node APIs to access the options passed in. Add an argument called `pluginOptions` to your `sourceNodes` function.
```

Comment:
```suggestion
Now you can use the second argument on Node APIs to access the options passed in. Add an argument called `pluginOptions` to your `sourceNodes` function.
```
Not sure I understand the first sentence.

Reply:
Yeah, I'll rework that...

```suggestion
Now the options you designated (like `previewMode: true`) will be passed into each of the Gatsby Node APIs like `sourceNodes`, making options accessible inside of Gatsby APIs. Add an argument called `pluginOptions` to your `sourceNodes` function.
```